lms/djangoapps/discussion_api/api.py

```diff
@@ -0,0 +1,104 @@
+"""
+Discussion API internal interface
+"""
+from collections import defaultdict
+from urllib import urlencode
+from urlparse import urlunsplit
+
+from django.core.urlresolvers import reverse
+
+from django_comment_client.utils import get_accessible_discussion_modules
+
+
+def get_course_topics(course, user, build_absolute_uri):
+    """
+    Return the course topic listing for the given course and user.
+
+    Parameters:
+
+    course: The course to get topics for
+    user: The requesting user, for access control
+    build_absolute_uri: a function that takes a relative URI and returns an
+        absolute URI
+
+    Returns:
+
+    A course topic listing dictionary; see discussion_api.views.CourseTopicViews
+    for more detail.
+    """
+    # TODO: This is temporary until the discussion_api thread list view is implemented
```

Comment:
I understand. While we want to ultimately change the thread_list_url to point to the new API, this is useful as is. If you are concerned about changing the thread_list_url, then I think it would make more sense to remove thread_list_url for this PR and add it later than to feature flag the whole API.